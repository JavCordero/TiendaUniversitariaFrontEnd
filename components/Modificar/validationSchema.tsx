import * as yup from "yup";

export default yup.object().shape({
  modificacion: yup.string()
  .matches(/^[1234567890]+$/, "Debe ser un valor numérico"),
  aumento: yup.string()
  .matches(/^[1234567890]+$/, "Debe ser un valor numérico"),
});