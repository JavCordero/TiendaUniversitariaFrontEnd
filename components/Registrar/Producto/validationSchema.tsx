import * as yup from "yup";

export default yup.object().shape({
  nombre: yup.string().required("El nombre es obligatorio"),
  precio: yup.string().required("El precio es obligatorio")
  .matches(/^[1234567890]+$/, "Debe ser un valor numérico"),
  stockActual: yup.string().required("EL stock actual es obligatorio")
  .matches(/^[1234567890]+$/, "Debe ser un valor numérico"),
  stockCritico: yup.string().required("EL stock crítico es obligatorio")
  .matches(/^[1234567890]+$/, "Debe ser un valor numérico"),
  categoria: yup.string().required("La categoría es obligatoria")
  .matches(/^[1234567890]+$/, "La categoría es obligatoria"),
  descripcion: yup.string().optional().max(255, "No debe superar los 255 caracteres"),
});