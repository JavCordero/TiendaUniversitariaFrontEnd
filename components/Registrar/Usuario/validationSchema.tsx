import * as yup from "yup";

export default yup.object().shape({
  nombre: yup.string().required("El nombre es obligatorio")
  .matches(/^[A-Za-záéíóúñÁÉÍÓÚ]+$/, "Deben ser caracteres alfabéticos"),
  apellido: yup.string().required("El apellido es obligatorio")
  .matches(/^[A-Za-záéíóúñÁÉÍÓÚ]+$/, "Deben ser caracteres alfabéticos"),
  email: yup.string().required("El correo electrónico es obligatorio").email("Debe ser un correo válido"),
  contrasenia: yup.string().required("La contraseña es obligatoria").min(8, "La contraseña debe tener mínimo 8 caracteres"),
  confirmContrasenia: yup.string().required("La contraseña es obligatoria")
  .oneOf([yup.ref("contrasenia"), null], "Las contraseñas deben coincidir"),
  rol: yup.string().required("El rol es obligatorio")
  .matches(/^[vendedor|administrador]+$/, "El rol es obligatorio"),
});