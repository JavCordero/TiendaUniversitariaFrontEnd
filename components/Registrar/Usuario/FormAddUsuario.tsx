import React from "react";
import router from "next/router";
import Image from "next/image";
import Link from "next/link";
import { Formik, Form, Field, ErrorMessage } from "formik";
import validationSchema from "./validationSchema";
import { registerUserAPI } from "../../../api/user";
import Swal from 'sweetalert2'

interface FormValues {
  nombre: string;
  apellido: string;
  email: string;
  contrasenia: string;
  confirmContrasenia: string;
  rol: string;
}
export const FormAddUsuario = (): JSX.Element => {
  const initialValues: FormValues = {
    nombre: "",
    apellido: "",
    email: "",
    contrasenia: "",
    confirmContrasenia: "",
    rol: "",
  };  

  const onSubmit = async (values: FormValues) => {
    try{
    const resp = await registerUserAPI(values.nombre, values.apellido, values.email, values.contrasenia, values.confirmContrasenia, values.rol);
    console.log(resp);

      if(resp.message == "Creado exitosamente"){
        Swal.fire(
          '¡Listo!',
          'Se ha registrado correctamente.',
          'success'
        )
        .then(function() {
          window.location.href = "/tienda/index";
      });

      } else {
        Swal.fire({
          icon: 'error',
          title: 'Oops...',
          text: 'Datos inválidos al registrarse, intente de nuevo.'
        })
      }
    } catch(error: any){

    }
  }

  return (
    <>
      <Formik initialValues={initialValues} onSubmit={onSubmit} validationSchema={validationSchema}>
        <Form id="registrar">
          <div className="row" id="fila2">
          </div>
          <div className="row" id="fila">
            <div id="register_user_left">  
              <div className="col-12">
                <label htmlFor="nombre" id="text">Nombre:</label>
                <Field className="form-control" name="nombre" type="text" placeholder="Nombre"/>
              </div>          
            </div>
            <div id="register_user_right">
              <div className="col-12">
                <label htmlFor="apellido" id="text">Apellido:</label>
                <Field className="form-control" name="apellido" type="text" placeholder="Apellido"/>
              </div>
            </div>
          </div>
          <div className="row" id="fila2">
            <div id="fila_unidad">  
              <div className="col-12">
                <ErrorMessage name="nombre" component="span" className="form__error"/>
              </div>          
            </div>
            <div id="fila_unidad">
              <div className="col-12">
                <ErrorMessage name="apellido" component="span" className="form__error"/>
              </div>
            </div>
          </div>
          <div className="row" id="fila">
            <div id="register_user_left">  
              <div className="col-12">
                <label htmlFor="email" id="text">Correo Electrónico:</label>
                <Field className="form-control" name="email" type="text" placeholder="correo@gmail.com"/>
              </div>          
            </div>
            <div id="register_user_right">
              <div className="col-12">
              <label id="text">Rol:</label>
                <Field as="select" name="rol" className="form-control" placeholder="Seleccione un rol">
                  <option>Seleccione un rol</option>
                  <option value="administrador">Administrador</option>
                  <option value="vendedor">Vendedor</option>
                </Field>
              </div>
            </div>
          </div>
          <div className="row" id="fila2">
            <div id="fila_unidad">  
              <div className="col-12">
                <ErrorMessage name="email" component="span" className="form__error"/>
              </div>          
            </div>
            <div id="fila_unidad">
              <div className="col-12">
                <ErrorMessage name="rol" component="span" className="form__error"/>
              </div>
            </div>
          </div>
          <div className="row" id="fila">
            <div id="register_user_left">     
              <div className="col-12">
                <label htmlFor="contrasenia" id="text">Contraseña:</label>
                <Field className="form-control" name="contrasenia" type="password" placeholder="********"/>
              </div>
            </div>
            <div id="register_user_right">
              <div className="col-12">
                <label htmlFor="confirmContrasenia" id="text">Confirmar Contraseña:</label>
                <Field className="form-control" name="confirmContrasenia" type="password" placeholder="********"/>                   
              </div>
            </div>
          </div>
          <div className="row" id="fila2">
            <div id="fila_unidad">  
              <div className="col-12">
                <ErrorMessage name="contrasenia" component="span" className="form__error"/>
              </div>          
            </div>
            <div id="fila_unidad">
              <div className="col-12">
                <ErrorMessage name="confirmContrasenia" component="span" className="form__error"/>
              </div>
            </div>
          </div>
          <div className="container-fluid" id="contenedor_boton">
            <button type="submit" id="boton_registrar" className="btn btn-primary btn-block btn-lg">Registrar</button>    
          </div>
        </Form>
      </Formik>
    </>
  );
};


