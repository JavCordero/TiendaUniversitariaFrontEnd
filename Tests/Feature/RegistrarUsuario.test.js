import React from 'react';
import { FormAddUsuario } from '../../components/Registrar/Usuario/FormAddUsuario.tsx';
import validationSchema from "../../components/Registrar/Usuario/validationSchema";
import {render, screen, waitFor} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { act } from 'react-dom/test-utils';

/**
* A tree containing both a providers and consumer can be rendered normally
*/

test('Debe renderizar el formulario de registro de usuario', () => {
  const { debug, container } = render(<FormAddUsuario />)
  
  debug();
  
  expect(container).toMatchSnapshot();

})

test('Envio de formulario', async () => {
    //falta el envio del formulario
    const handleSubmit = jest.fn();
    render(<FormAddUsuario onSubmit={handleSubmit} validationSchema={validationSchema} />)
  
    /*
    //cambia valores de los inputs
    act(() =>{
        userEvent.type(screen.getByPlaceholderText("Nombre"), 'Luciano');
        userEvent.type(screen.getByPlaceholderText("Apellido"), 'Larama');
        userEvent.type(screen.getByPlaceholderText("correo@gmail.com"), 'luciano@ucn.cl');
        userEvent.type(screen.getByPlaceholderText("Seleccione un rol"), 'administrador');
    })

    userEvent.click(screen.getByRole('button', {name: /submit/i}));  
    */

    //cambia valores de los inputs
    /*
    userEvent.type(screen.getByPlaceholderText("Nombre"), 'Luciano');
    userEvent.type(screen.getByPlaceholderText("Apellido"), 'Larama');
    userEvent.type(screen.getByPlaceholderText("correo@gmail.com"), 'luciano@ucn.cl');
    userEvent.type(screen.getByPlaceholderText("Seleccione un rol"), 'administrador');
    */

    
    userEvent.click(screen.getByRole('button', {name: /submit/i}));  

    await waitFor(() =>
    
      expect(handleSubmit).toHaveBeenCalledWith({
        nombre: 'Luciano',
        apellido: 'Larama',
        email: 'luciano@ucn.cl',
        //contrasenia: "123123123",
        //confirmContrasenia: "123123123",
        contrasenia: "",
        confirmContrasenia: "",
        rol: 'administrador',
      }),
    )
    
})