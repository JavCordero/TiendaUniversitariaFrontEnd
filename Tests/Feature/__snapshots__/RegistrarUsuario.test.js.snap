// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe renderizar el formulario de registro de usuario 1`] = `
<div>
  <form
    action="#"
    id="registrar"
  >
    <div
      class="row"
      id="fila2"
    />
    <div
      class="row"
      id="fila"
    >
      <div
        id="register_user_left"
      >
        <div
          class="col-12"
        >
          <label
            for="nombre"
            id="text"
          >
            Nombre:
          </label>
          <input
            class="form-control"
            name="nombre"
            placeholder="Nombre"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        id="register_user_right"
      >
        <div
          class="col-12"
        >
          <label
            for="apellido"
            id="text"
          >
            Apellido:
          </label>
          <input
            class="form-control"
            name="apellido"
            placeholder="Apellido"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="row"
      id="fila2"
    >
      <div
        id="fila_unidad"
      >
        <div
          class="col-12"
        />
      </div>
      <div
        id="fila_unidad"
      >
        <div
          class="col-12"
        />
      </div>
    </div>
    <div
      class="row"
      id="fila"
    >
      <div
        id="register_user_left"
      >
        <div
          class="col-12"
        >
          <label
            for="email"
            id="text"
          >
            Correo Electrónico:
          </label>
          <input
            class="form-control"
            name="email"
            placeholder="correo@gmail.com"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        id="register_user_right"
      >
        <div
          class="col-12"
        >
          <label
            id="text"
          >
            Rol:
          </label>
          <select
            class="form-control"
            name="rol"
            placeholder="Seleccione un rol"
          >
            <option>
              Seleccione un rol
            </option>
            <option
              value="administrador"
            >
              Administrador
            </option>
            <option
              value="vendedor"
            >
              Vendedor
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="row"
      id="fila2"
    >
      <div
        id="fila_unidad"
      >
        <div
          class="col-12"
        />
      </div>
      <div
        id="fila_unidad"
      >
        <div
          class="col-12"
        />
      </div>
    </div>
    <div
      class="row"
      id="fila"
    >
      <div
        id="register_user_left"
      >
        <div
          class="col-12"
        >
          <label
            for="contrasenia"
            id="text"
          >
            Contraseña:
          </label>
          <input
            class="form-control"
            name="contrasenia"
            placeholder="********"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        id="register_user_right"
      >
        <div
          class="col-12"
        >
          <label
            for="confirmContrasenia"
            id="text"
          >
            Confirmar Contraseña:
          </label>
          <input
            class="form-control"
            name="confirmContrasenia"
            placeholder="********"
            type="password"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="row"
      id="fila2"
    >
      <div
        id="fila_unidad"
      >
        <div
          class="col-12"
        />
      </div>
      <div
        id="fila_unidad"
      >
        <div
          class="col-12"
        />
      </div>
    </div>
    <div
      class="container-fluid"
      id="contenedor_boton"
    >
      <button
        class="btn btn-primary btn-block btn-lg"
        id="boton_registrar"
        type="submit"
      >
        Registrar
      </button>
    </div>
  </form>
</div>
`;
